apps/emqx_authn/test/emqx_authn_api_SUITE.erl

```diff
@@ -164,10 +164,38 @@ test_authenticator(PathPrefix) ->
                      post,
                      uri(PathPrefix ++ [?CONF_NS]),
                      ValidConfig0),
-    {ok, 200, _} = request(
+    {ok, 200, Res} = request(
                      get,
                      uri(PathPrefix ++ [?CONF_NS, "password-based:http"])),
-
+    {ok, RList} = emqx_json:safe_decode(Res),
+    Snd = fun ({_, Val}) -> Val end,
+    LookupVal =
+        fun(FixVal) ->
+            fun(List, RestJson) ->
+                case List of
+                    [Name] -> Snd(lists:keyfind(Name, 1, RestJson));
+                    [Name | NS] -> (FixVal(FixVal))(NS, Snd(lists:keyfind(Name, 1, RestJson)))
+                end
+            end
+        end,
+    LookFun = fun (List) -> (LookupVal(LookupVal))(List, RList) end,
```

Comment:
Make it more readable if possible, as the code is to be maintained by other developers.